### ISSUE TITLE
Batching affects predictions of EncoderASR and EncoderDecoderASR

### ISSUE THREAD

**Issue by olvb** [speechbrain/speechbrain#1597]
Batching together audios of different duration affects the prediction returned by `EncoderASR` and `EncoderDecoderASR`, ie `prediction(Batch[wav_A, wav_B]) != [prediction(Batch[wav_A]), prediction(Batch[wav_B])]`

With `EncoderDecoderASR`, it really seems as if the `wav_lens` are not taken into account at some point, as batching an audio with another much longer audio results in a much longer prediction.

Colab reproducing the issue: https://colab.research.google.com/drive/1glxrATirT5zQ11Lj9tFDBqXi25XzaSLj?usp=sharing

**Comment by olvb** [comment]
One cause is that we are not providing attention masks to `Wav2Vec2Model` when passing the padded signals, but this is needed for some models (cf https://huggingface.co/docs/transformers/model_doc/wav2vec2#transformers.Wav2Vec2FeatureExtractor.return_attention_mask).

I suppose one way to fix that would be to add an additional `lengths` argument to the `forward()` method of `HuggingFaceWav2Vec2` and use it to compute the attention masks. But this probably wouldn't be enough because we still wouldn't be taking into account the padding when normalizing the inputs and outputs with `F.layer_norm()`

**Comment by anautsch** [comment]
Will be resolved in PR #1596 :)
We will use the SB layer norm there which respects wav_lens.

**Comment by olvb** [comment]
Great news, thanks!

**Comment by Adel-Moumen** [comment]
Hello, 

Thanks for opening this issue.

After working on that, we realised that this is not an issue. There is just a little variability in the performance when we change the batch size due to tail effects. Fixing that is possible, but not convenient because the users would have to feed in input a mask and return another mask for every basic operation involving tail effects, such as STFT, convolutions, bidirectional RNNs, etc. This is super annoying and prevents users to use things like the standard sequential operator, conv, etc. And this is why we are not planning to fix it unfortunately. As a consequence, I am closing this issue. 

Best, 
Adel
